### PR TITLE
(2/2) Implement skeleton for getTransactionHistory

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement a skeleton of `getTransactionHistory` and tests inside of XpringClient.

JS: https://github.com/xpring-eng/Xpring-JS/pull/212
Java: https://github.com/xpring-eng/xpring4j/pull/116
Swift: https://github.com/xpring-eng/XpringKit/pull/124

### Context of Change

rippled recently implemented this RPC which we'd like to use.

Note that this functionality is not exposed in `XpringClient`, so this is still internal only. Currently `Transaction` objects are just empty containers. 

A future PR will:
- Update `Transaction` to have fields and to set them correctly from the protocol buffers
- Expose this as a public API

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

N/A - still internal only functionality. 

## Test Plan

New unit tests provided for CI.

### CODE REVIEW

File: Tests/DefaultXpringClientTest.swift [hunk 1]
```diff
@@ -418,6 +418,66 @@ final class DefaultXpringClientTest: XCTestCase {
     }
   }
 
+  // MARK: - TransactionHistory
```
Comment (hbergren) [inline Tests/DefaultXpringClientTest.swift]: What does this `MARK` mean?

File: Tests/DefaultXpringClientTest.swift [hunk 2]
```diff
@@ -418,6 +418,66 @@ final class DefaultXpringClientTest: XCTestCase {
...
+  func testTransactionHistoryWithSuccess() {
+    // GIVEN a Xpring client which will successfully return transactionHistory mocked network call.
```
Comment (hbergren) [inline Tests/DefaultXpringClientTest.swift]:
```suggestion
    // GIVEN a Xpring client which will successfully return a transactionHistory mocked network call.
```

File: Tests/Helpers/Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse+Test.swift [hunk 3]
```diff
@@ -1,9 +1,22 @@
 import XpringKit
 
+// TODO(keefertaylor): Refactor Org_Xrpl_Rpc_V1_AccountAddress to be a static test helper object to avoid repetition.
 extension Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse {
   static let testTransactionHistoryResponse = Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse.with {
     $0.account = Org_Xrpl_Rpc_V1_AccountAddress.with {
       $0.address = .testAddress
     }
+
+    $0.transactions = [
```
Comment (hbergren) [inline Tests/Helpers/Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse+Test.swift]: What does the `$0` here mean?

Comment (amiecorso) [reply Tests/Helpers/Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse+Test.swift]: yeah I'd love to know this too

File: Tests/DefaultXpringClientTest.swift (hunk 1 above)
Comment (amiecorso) [reply Tests/DefaultXpringClientTest.swift]: keefer explains this here: https://github.com/xpring-eng/XpringKit/pull/116#discussion_r387110813

File: Tests/Helpers/Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse+Test.swift (hunk 3 above)
Comment (keefertaylor) [reply Tests/Helpers/Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse+Test.swift]: This is kinda complicated. 

this method is defined as:
```
    public static func with(_ populator: (inout Self) throws -> ()) rethrows -> Self
````

What this says is: 
- `with` is a static method  (`public static func with(`)
- which takes a closure (you could also call this an anonymous function or lambda in other languages) (`_ populator: (inout Self) throws -> ())
- This block takes an inout parameter of type self
- And the whole function returns an instance of `Self` (`-> Self`)

So, we could write something like this (long form):
```
Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse.with((populator: Self) in 
  populator.transactions = <...>
)
````

Having the last parameter of a method be a closure is pretty common in swift, so we actually have special syntax when we write outside of the method ("trailing closure syntax"):

```
Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse.with() { (populator: Self) in 
  populator.transactions = <...>
}
```

Trailing closure syntax also lets us drop the `()` for brevity if there are no other parameters. And we can drop the type annotation in the parameter to the closure:
```
Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse.with { populator in 
  populator.transactions = <...>
}
```

Swift also lets us drop the names of parameters and refer to them as `$<param number` in closures, so we can take this one step further:

```
Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse.with { 
  $0.transactions = <...>
}
```

(This is in fact the recommended syntax from the Swift guide: https://github.com/apple/swift-protobuf#quick-start)

Comment (keefertaylor) [reply Tests/Helpers/Org_Xrpl_Rpc_V1_GetAccountTransactionHistoryResponse+Test.swift]: In general, Swift has some gnarly closure syntax if you don't know what you're looking at. The definitive reference is: 
https://fuckingclosuresyntax.com/

You might also find this guide useful: https://www.hackingwithswift.com/example-code/language/what-is-trailing-closure-syntax

**Pull request merged**